### PR TITLE
Vision Transformer for Brain MRI Classification (Alzheimers of Normal)

### PR DESCRIPTION
**Pull request by gp-rgb**
Vision Transformer trained on a binary classification exercise with MRI image data of healthy brains (NC) and those with varying presentations of alzheimers (AD). The code provided is a complete solution to problem 6.

The files included in this pull request are:
`README.md`: A report-style description of the problem, the model and the development process.
`dataset.py`: Python module that provides necessary functions to load the ADNI dataset into a tensorflow-compatible format. 
`demo.py`: Demonstration script that runs the model on several testing images and shows predicted and true classes.
`modules.py`: Python module that provides class and function declarations necessary to initialise a Vision Transformer model.
`myenv.txt`: Text file listing all environment variables, that is able to be called by `pip` to install all packages required to run the solution code.
`predict.py`: Classifies a single image specified by a user-defined path.
`train.py`: Training script that produced a Vision Transformer model with greater than 70% testing accuracy on the ADNI Dataset, in under 10 epochs.

### CODE REVIEW

**Comment by aryaman-sh** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:


Commit messages:

Code:

Functionality/performance:

General comments:

**Comment by aryaman-sh** [comment]
**This is an initial inspection, no action is required at this point**

**Difficulty** 
- Hard

**Readme**
- Missing references section

**Commit messages**
- good, detailed

**Code**
- good comments
- good design

**Functionality/performance**
- 70% validation accuracy
- Good plots
- Can show hyperparameter tuning steps

**Comment by aryaman-sh** [comment]
**Feedback update**
- Added references section to README

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required (Feedback done)
Request Description good

**Comment by shakes76** [comment]
Can't merge because topic-recognition branch README has been deleted. Can you restore and I can merge your PR. This does not affect your grade.

**Comment by wangzhaomxy** [comment]
feedback attempt.
